Controllers films users

ArsenyIshtubaev opened the pull request.

ArsenyIshtubaev committed:
-first commit

ArsenyIshtubaev committed:
-feat:
Спринт8 - создать каркас Spring Boot приложения Filmorate

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,60 @@
+@RestController
+@Slf4j
+@RequestMapping("/films")
+public class FilmController {
```
обрати внимание на то, как описаны эти контроллеры в тз, в обоих перечислены три одинаковых метода обновить, создать получить все записи. Попробуй применить тут знания о абстрактных классах.

ArsenyIshtubaev replied:
Не совсем понял как правильно тут применить абстракцию. Добавил абстрактный класс Controller, но логика методов в классах наследниках получилась уникальная, т.к. есть метод валидации

DyndaNE commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -0,0 +1,198 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+class FilmorateApplicationTests {
```
Молодец, что сразу добавил тесты!

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,32 @@
+@Setter
+@AllArgsConstructor
+@Builder
+public class User {
```
молодец, что тут использовал другие аннотации вместо Data

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,32 @@
+    private String email;
+    private String login;
+    private String name;
+    private final LocalDate birthday;
```
попробуй добавить аннотации валидации к полям например
@NotBlank
@Size(min = 1, max = 200)
@NotNull
@Min(1)
@Email 

к обоим классам

ArsenyIshtubaev replied:
Аннотации валидации добавил и отредактировал тесты под данную реализацию валидации

ArsenyIshtubaev committed:
-fix:
Спринт8 (2)
- добавить аннотации валидации к полям
- добавить абстрактный класс Controller

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/Controller.java:
```diff
@@ -0,0 +1,18 @@
+    abstract T create(@Valid @RequestBody T object) throws ValidationException;
+
+
+    abstract T update(@Valid @RequestBody T object) throws ValidationException;
```
просто стоило добавить сюда абстрактный метода валидации например
abstract void validate(T data);

логика его вызова одинакова в обоих методах, а в контроллерах переопределяешь его

ArsenyIshtubaev committed:
-fix:
Спринт 8 (3)
-добавить абстрактный метод validate и переопределить его в классах наследниках
